File: 20160628/20160628_1.cc
#include<stdio.h>

class Solution {
public:
	bool isPerfectSquare(int num)
	{
		long r = num;
		while (r*r > num)
			r = (r + num /r) / 2;
		return r*r == num;
	}
};

int main()
{
	Solution s = Solution();
	printf("%5s %10s %10s\n", "input", "Expected", "Result");	
	printf("%5d %10s %10s\n", 16, "T", s.isPerfectSquare(16)?"T":"F");
	printf("%5s %10s %10s\n", "input", "Expected", "Result");
	printf("%5d %10s %10s\n", 14, "F", s.isPerfectSquare(14)?"T":"F");
	return 0;
}

File: 20160628/20160628_1.py
class Solution:
    def isPerfectSquare(self, num):
        """
        :type num: int
        :rtype: bool
        """
        if num == 1:
            return True
        left = 1
        right = num
        while left <= right:
            mid = (left + right) // 2
            if mid * mid > num:
                right = mid - 1
            elif mid * mid < num:
                left = mid + 1
            else:
                return True
        return False

s = Solution()
print "%5s %10s %10s" % ("input", "Expected", "Result")
print "%5d %10s %10s" % (16, "True", s.isPerfectSquare(16))
print "%5s %10s %10s" % ("input", "Expected", "Result")
print "%5d %10s %10s" % (14, "False", s.isPerfectSquare(14))
